pkg/catalog/catalog.go

```diff
@@ -156,6 +156,23 @@ func (c *Catalog) WithReleaseFilter(filter filtering.Filter) *Catalog {
 	return c
 }
 
+func (c *Catalog) WithReleases(names []string) *Catalog {
+	if len(names) == 0 {
+		return c
+	}
+
+	releases := c.Releases.Items
+	c.Releases.Items = []*cross.Release{}
+
+	for _, cross := range releases {
+		if slices.Contains(names, cross.Name) {
+			c.Releases.Items = append(c.Releases.Items, cross)
+		}
+	}
+
+	return c
+}
```

Comment:
Not sure I like that. It's a mutating function disguised into a function that would be returning a filtered copy. I would either rename it to something like `KeepOnlyReleases()` and remove the return value, or more ideally make it deeply clone the catalog.

Reply:
I did not change the names but I stopped returning the reference, making the API non-chainable and breaking any possible illusion that any cloning is happening.